### ISSUE TITLE
Dataset preview fails if non-English letters are present: "Import Dataset → From Text (readr)" 

### ISSUE THREAD

**Issue by GegznaV** [rstudio/rstudio#2675]
This issue is similar to #2674

### System details

    RStudio Edition : Desktop
    RStudio Version : 1.1.447
    OS Version :      Windows 10
    R Version :       3.4.4 (2018-03-15)

### Steps to reproduce the problem

1. Create a text file that contains at least one non-English letter ("ą", "ž", etc.) in the path name or file name, lile "c:/užduotis-1/iris.txt". This code does this in Windows (in Mac and Linux the line for setting the locale should be changed appropriately):

```r
Sys.setlocale(locale = "Lithuanian")
data("iris")
dir.create("c:/užduotis-1", recursive = TRUE)
write.csv(iris, file = "c:/užduotis-1/iris.txt")
```

2. Choose the method to import data: 
	* RStudio → window "Environment"  → button "Import Dataset" → From Text (readr)

3. Trying to open the file "c:/užduotis-1/iris.txt" results in an error message and failure to preview a dataset.

![image](https://user-images.githubusercontent.com/12725868/39063245-0d05b43c-44d3-11e8-80f1-ee4300bedf38.png)
In the figure, red "X" indicates the place where path/file name is distorted (no letter "ž" is present), and green "v" indicates places where the path/file name is correct.

Note: the code generated by the "import using `readr`" window works as expected.

### Describe the problem in detail

Dataset preview feature does no work in "*Import Dataset" → From Text (readr)*" if the pathname or the filename of the file with data contains non-English letter(s).

Note: when "c:/užduotis-1/iris.txt" is changed into "c:/uzduotis-1/iris.txt" (i.e., "ž" →"z,") the file import menu works as expected". 

### Describe the behavior you expected

It should be possible to use the preview feature even if non-English letters exist in the file or path name.

**Comment by ronblum** [comment]
Thank you for reporting this issue, @GegznaV. We'll review it as we work on ongoing development of RStudio.

**Comment by ronblum** [comment]
Note: This also occurs with base, Excel, etc. Also, there was a somewhat similar issue, #1868, in which non-UTF-8 characters in the file name caused problems with `file.edit()`.

**Comment by GegznaV** [comment]
Seems to be solved in 1.4.1103
